agents/tests/test_agent.py

```diff
@@ -589,3 +590,63 @@ def test_maybe_add_to_queue_logs_bad_input():
 
     with agent.queue.mutex:
         assert len(agent.queue.queue) == 0
+
+
+def test_status_to_dict_keys():
+    """
+    Verify that status_to_dict() returns all expected keys
+    in the status dictionary.
+    Uses mock to isolate from metadata collection implementation.
+    """
+    agent = MockAgent(port=12345)
+    with mock.patch.object(agent, 'collect_server_metadata') as mock_collect:
+        agent.os_type = 'linux'
+        agent.hostname = 'test-host'
+        agent.ip = '127.0.0.1'
+        agent.server_name = 'dns'
+        agent.uptime = 12345
+        agent.timestamp = '2025-06-03 20:00:00'
+        agent.healthy = True
+
+        result = agent.status_to_dict()
+
+        required_keys = set([
+            'os',
+            'hostname',
+            'ip',
+            'server_name',
+            'uptime',
+            'timestamp',
+            'healthy',
+            ])
+        msg_mock = 'Expected collect_server_metadata() to be called only once'
+        mock_collect.assert_called_once(), msg_mock
+
+        msg_keys = (
+            'Expected status_to_dict() keys to match required: {0}'.format(
+                required_keys
+            )
+        )
+        assert set(result.keys()) == required_keys, msg_keys
```

Comment:
I don't understand why you're patching `collect_server_metadata` if you're not actually calling it? And why `mock_collect.assert_called_once()` passes